odoo/cli/populate.py

```diff
@@ -22,38 +22,68 @@ def run(self, cmdargs):
         group.add_option("--models",
                          dest='populate_models',
                          help="Comma separated list of model or pattern (fnmatch)")
+        group.add_option("--profile",
+                         dest='profiling_enabled', action="store_true",
+                         help="Specify if you want to profile records population.",
+                         default=False)
+        group.add_option("--rollback",
+                         dest='populate_rollback', action="store_true",
+                         help="Specify if you want to rollback database population.",
+                         default=False)
         parser.add_option_group(group)
         opt = odoo.tools.config.parse_config(cmdargs)
         populate_models = opt.populate_models and set(opt.populate_models.split(','))
         population_size = opt.population_size
+        profiling_enabled = opt.profiling_enabled
         dbname = odoo.tools.config['db_name']
         registry = odoo.registry(dbname)
         with registry.cursor() as cr:
             env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
-            self.populate(env, population_size, populate_models)
-
+            self.populate(
+                env, population_size, populate_models,
+                profiling_enabled=profiling_enabled,
+                rollback=opt.populate_rollback)
 
     @classmethod
-    def populate(cls, env, size, model_patterns=False):
+    def populate(cls, env, size, model_patterns=False, profiling_enabled=False, rollback=False):
         registry = env.registry
         populated_models = None
         try:
             registry.populated_models = {}  # todo master, initialize with already populated models
             ordered_models = cls._get_ordered_models(env, model_patterns)
 
             _logger.log(25, 'Populating database')
+            if rollback:
+                env.cr.execute('SAVEPOINT pre_populate')
             for model in ordered_models:
                 _logger.info('Populating database for model %s', model._name)
                 t0 = time.time()
-                registry.populated_models[model._name] = model._populate(size).ids
+                if rollback:
+                    model = model.with_context(populate_commit=False)
+                if profiling_enabled:
+                    with odoo.tools.profiler.Profiler(
+                        description=f'{model} {size}',
+                        db=env.cr.dbname
+                    ):
+                        registry.populated_models[model._name] = model._populate(size).ids
+                else:
+                    registry.populated_models[model._name] = model._populate(size).ids
```

Comment:
Not sure what is the best solution but we could define the context manager outside and have a nullcontext if profiling is not active